conftest.py

```diff
@@ -0,0 +1,135 @@
+import os
+from uuid import uuid1
+
+import pytest
+
+from hub.constants import (MIN_LOCAL_CACHE_SIZE, MIN_MEMORY_CACHE_SIZE,
+                           PYTEST_LOCAL_PROVIDER_BASE_ROOT,
+                           PYTEST_MEMORY_PROVIDER_BASE_ROOT,
+                           PYTEST_S3_PROVIDER_BASE_ROOT)
+from hub.core.storage import LocalProvider, MemoryProvider, S3Provider
+from hub.util.cache_chain import get_cache_chain
+
+
+def _skip_if_none(val):
+    if val is None:
+        pytest.skip()
+
+
+def _is_opt_true(request, opt):
+    return request.config.getoption(opt)
+
+
+def pytest_addoption(parser):
+    parser.addoption(
+        "--memory-skip",
+        action="store_true",
+        help="Tests using the `memory_provider` fixture will be skipped. Tests using the `storage` fixture will be skipped if called with \
+                `MemoryProvider`.",
+    )
+    parser.addoption(
+        "--local",
+        action="store_true",
+        help="Tests using the `storage`/`local_provider` fixtures will run with `LocalProvider`.",
+    )
+    parser.addoption(
+        "--s3",
+        action="store_true",
+        help="Tests using the `storage`/`s3_provider` fixtures will run with `S3Provider`.",
+    )
+    parser.addoption(
+        "--cache-chains",
+        action="store_true",
+        help="Tests using the `storage` fixture may run with combinations of all enabled providers \
+                in cache chains. For example, if the option `--s3` is not provided, all cache chains that use `S3Provider` are skipped.",
+    )
+    parser.addoption(
+        "--cache-chains-only",
+        action="store_true",
+        help="Force enables `--cache-chains`. `storage` fixture only returns cache chains. For example, if `--s3` is provided, \
+            `storage` will never be just `S3Provider`.",
+    )
+
+
+@pytest.fixture(scope="session")
+def memory_storage(request):
+    if not _is_opt_true(request, "--memory-skip"):
+        root = PYTEST_MEMORY_PROVIDER_BASE_ROOT
+        return MemoryProvider(root)
+
+
+@pytest.fixture(scope="session")
+def local_storage(request):
+    if _is_opt_true(request, "--local"):
+        # TODO: root as option
+        root = PYTEST_LOCAL_PROVIDER_BASE_ROOT
+        return LocalProvider(root)
+
+
+@pytest.fixture(scope="session")
+def s3_storage(request):
+    if _is_opt_true(request, "--s3"):
+        # TODO: root as option
```

Comment:
Make sure to put this in the backlog if it's out-of-scope. It's kinda important for S3, since concurrent instances' tests will interfere with each other. We should at least tack on a hash or timestamp to the S3 base so this doesn't happen.

Reply:
there is a session UUID

Reply:
but it's for the `current_test_name` that way when you look at the s3 test bucket you don't just see a ton of UUIDs but you see the test names & then within there you see the UUIDs

Reply:
im making this session UUID constant & printing it out before & after tests start